demos/mqtt/mqtt_demo_basic_tls/mqtt_demo_basic_tls.c

```diff
@@ -0,0 +1,466 @@
+ *
+ * @return EXIT_SUCCESS if an MQTT session is established; EXIT_FAILURE otherwise.
+ */
+static int establishMqttSession( MQTTContext_t * pContext, SSL * pSslContext )
+{
+    int status = EXIT_SUCCESS;
+    MQTTStatus_t mqttStatus;
+    MQTTConnectInfo_t connectInfo;
+    MQTTTransportInterface_t transport;
+    MQTTFixedBuffer_t networkBuffer;
+    MQTTApplicationCallbacks_t callbacks;
+
+    /* The network buffer must remain valid for the lifetime of the MQTT context. */
+    static uint8_t buffer[ NETWORK_BUFFER_SIZE ];
+
+    /* Initialize MQTT context. */
+    transport.networkContext = pSslContext;
+    transport.send = transportSend;
+    transport.recv = transportRecv;
+
+    networkBuffer.pBuffer = buffer;
+    networkBuffer.size = NETWORK_BUFFER_SIZE;
+
+    callbacks.appCallback = eventCallback;
+    callbacks.getTime = getTime;
+
+    MQTT_Init( pContext, &transport, &callbacks, &networkBuffer );
+
+    /* Establish MQTT session with a CONNECT packet. */
+    connectInfo.cleanSession = true;
+    connectInfo.pClientIdentifier = CLIENT_IDENTIFIER;
+    connectInfo.clientIdentifierLength = CLIENT_IDENTIFIER_LENGTH;
+    connectInfo.keepAliveSeconds = 0;
+    connectInfo.pUserName = NULL;
+    connectInfo.userNameLength = 0;
+    connectInfo.pPassword = NULL;
+    connectInfo.passwordLength = 0;
+
+    mqttStatus = MQTT_Connect( pContext, &connectInfo, NULL, NULL );
+
+    if( mqttStatus != MQTTSuccess )
+    {
+        status = EXIT_FAILURE;
+    }
+
+    return status;
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Close an MQTT session by sending MQTT DISCONNECT.
+ *
+ * @param[in] pContext MQTT context.
+ *
+ * @return EXIT_SUCCESS if DISCONNECT was successfully sent; EXIT_FAILURE otherwise.
+ */
+static int disconnectMqttSession( MQTTContext_t * pContext )
+{
+    int status = EXIT_SUCCESS;
+
+    MQTTStatus_t mqttStatus = MQTT_Disconnect( pContext );
+
+    if( mqttStatus != MQTTSuccess )
+    {
+        status = EXIT_FAILURE;
+    }
+
+    return status;
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Entry point of demo.
+ */
+int main( int argc, char ** argv )
+{
+    bool mqttSessionEstablished = false;
+    int status;
+    MQTTContext_t context;
+    SSL * pSslContext = NULL;
+
+    /* Establish TCP+TLS connection and MQTT session. */
+    int tcpSocket = connectToServer( SERVER, PORT );
+
+    if( tcpSocket == -1 )
+    {
+        status = EXIT_FAILURE;
+    }
+    else
+    {
+        status = EXIT_SUCCESS;
+    }
+
+    if( status == EXIT_SUCCESS )
+    {
+        pSslContext = tlsSetup( tcpSocket );
+
+        if( pSslContext == NULL )
+        {
+            status = EXIT_FAILURE;
+        }
+    }
+
+    if( status == EXIT_SUCCESS )
+    {
+        status = establishMqttSession( &context, pSslContext );
+
+        if( status == EXIT_SUCCESS )
+        {
+            mqttSessionEstablished = true;
+        }
+    }
+
```

Comment:
Please add comment here mentioning it is not a complete demo yet but shows trasport layer setup, so everyone else understands.

Reply:
done